acp/acp.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+/*
+Package acp utilizes the sourcehub acp module to bring the functionality to defradb, this package also helps
+avoid the leakage of direct sourcehub references through out the code base, and eases in swapping
+between local embedded use case and a more global on sourcehub use case.
+*/
+
+package acp
+
+import (
+	"context"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/logging"
+)
+
+var (
+	log = logging.MustNewLogger("acp")
+
+	NoACPModule = immutable.None[ACPModule]()
+)
+
+// ACPModule is the interface to all types of access control modules there might exist.
+//
+// An embedded local implementation might restrict the implementation to not make remote calls.
```

Comment:
todo: It is not clear what you mean by embedded here - can you please clarify?

This whole line feels a bit odd too, but every time I try and suggest something I feel more unclear about it - maybe remove it, perhaps along with the remote stuff below?  There is nothing in the interface that suggests that it may or may not make remote calls, and writing about it in this way on the interface declaration might be causing more confusion than it is removing.

Reply:
- [ ] I will remove the below verbose documentation, but keep title (first line)